ProtoSDK/MS-KILE/KileDecrypter.cs

```diff
@@ -241,18 +241,18 @@ public void ApExchange(byte[] apReq, byte[] apRep)
         {
             if (apReq == null)
             {
-                throw new ArgumentNullException("apReq");
+                throw new ArgumentNullException(nameof(apReq));
             }
             if (apRep == null)
             {
-                throw new ArgumentNullException("apRep");
+                throw new ArgumentNullException(nameof(apRep));
             }
 
-            KileApRequest apReqPdu = new KileApRequest(kileDecoder.serverContext);
+            var apReqPdu = new KileApRequest(kileDecoder.serverContext);
```

Comment:
We explicitly declare the type of variable, this is by design.

Reply:
Look at the conventions [here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions)

> Use implicit typing for local variables when the type of the variable is obvious from the right side of the assignment, or when the precise type is not important.